test/functional/tests/lazy_writes/test_parallel_core_flushing.py

```diff
@@ -0,0 +1,116 @@
+from test_tools.dd import Dd
+from test_utils.filesystem.file import File
+from test_utils.output import CmdException
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_parallel_core_flushing():
+    """
+        title: Test for parallel cached volume flushing.
+        description: Test checks whether all cores attached to one cache instance are flushed
+                     in parallel after executing flush cache command.
+        pass_criteria:
+          - all cores should flushed in parallel
+          - checksums for cores and core devices should be identical
+    """
+    with TestRun.step("Prepare devices."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(8, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks['core']
+        core_disk.create_partitions([Size(2, Unit.GibiByte)] * 4)
+        core_devices = core_disk.partitions
+
+    with TestRun.step("Start cache in Write-Back mode and add cores."):
+        cache = casadm.start_cache(cache_dev, cache_mode=CacheMode.WB)
+        cores = [cache.add_core(dev) for dev in core_devices]
+        cache.set_cleaning_policy(CleaningPolicy.nop)
+
+    with TestRun.step("Run IO on each CAS device."):
+        for core in cores:
+            Dd().output(core.path)\
+                .input("/dev/urandom")\
+                .block_size(Size(1, Unit.Blocks4096))\
+                .run()
+
+    with TestRun.step("Check if occupancy of all CAS devices increased "
+                      "and there is dirty data on them."):
+        proper_stats = (0.9 * Size(2, Unit.GibiByte)).get_value()
+        occupancy = core.get_occupancy().get_value()
+        dirty = core.get_dirty_blocks().get_value()
+        for core in cores:
+            if occupancy > proper_stats and dirty > proper_stats:
+                TestRun.LOGGER.info(f"Stats are as expected for core {core.core_id}.")
+            else:
+                TestRun.fail(f"Stats are not as expected for core {core.core_id}\n"
+                             f"Occupancy: {occupancy}\n"
+                             f"Dirty: {dirty}\n"
+                             f"Proper stats: {proper_stats}")
+
+    with TestRun.step("In background run flush cache command."):
+        TestRun.executor.run_in_background(cli.flush_cache_cmd(str(cache.cache_id)))
+
+    with TestRun.step("Check whether all CAS devices are in 'Flushing' state and wait for finish."):
+        for core in cores:
+            wait_for_flushing_start(cache, core)
+
+        percentages = [0] * len(cores)
+        try:
+            while not all(p == 100 for p in percentages):
+                for i, core in enumerate(cores):
+                    if percentages[i] != 100:
+                        percentage = get_flushing_progress(cache.cache_id, core.core_id)
```

Comment:
As soon as one of cores is flushed, `get_flushing_progress()` will throw an exception. The loop will be interrupted and the flushing progress bar for other cores won't be tested. I suggest putting only `get_flushing_progress()` into `try` block.

Reply:
You can even go one step further and implement function that would return a list with flushing progress for each core. 

```
def get_progress(cache, cores):
    progress = [0] * len(cores)
    for i, core in enumerate(cores)::
        try:
            progress[i] = get_flushing_progress(cache.cache_id, core.core_id)
        except CmdException:
             progress[i] = 100
    return progress
```

usage:

```
percentages = [0] * len(cores)
while TestRun.executor.check_if_process_exists(pid):
    current_values = get_progress(cache, cores)

    for old, new, core zip(percentages, current_values, cores)
        if old > new:
            TestRun.fail(f"Core {core.id}: progress decreased from {old} to {new}")

    percentage = current_values
```